tests/integration_tests/functional/test_snapshot_version.py

```diff
@@ -123,3 +151,69 @@ def test_restore_in_past_versions(bin_cloner_path):
 
         exit_code, _, _ = ssh_connection.execute_command("sleep 1 && sync")
         assert exit_code == 0
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_create_with_past_version(test_microvm_with_ssh, network_config):
+    """Test scenario: restore in previous versions with too many devices."""
+    test_microvm = test_microvm_with_ssh
+
+    # Create and start a microVM with (`FC_V0_23_MAX_DEVICES_ATTACHED` - 1)
+    # network devices.
+    devices_no = FC_V0_23_MAX_DEVICES_ATTACHED - 1
+    _create_and_start_microvm_with_net_devices(test_microvm,
+                                               network_config,
+                                               devices_no)
+
+    snapshot_builder = SnapshotBuilder(test_microvm)
+    # Create directory and files for saving snapshot state and memory.
+    _snapshot_dir = snapshot_builder.create_snapshot_dir()
+    # Pause and create a snapshot of the microVM. Firecracker v0.23 allowed a
+    # maximum of `FC_V0_23_MAX_DEVICES_ATTACHED` virtio devices at a time.
+    # This microVM has `FC_V0_23_MAX_DEVICES_ATTACHED` devices, including the
+    # rootfs, so snapshotting should succeed.
+    test_microvm.pause_to_snapshot(
+        mem_file_path="/snapshot/vm.mem",
+        snapshot_path="/snapshot/vm.vmstate",
+        diff=False,
+        version="0.23.0")
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64",
+    reason="Not supported yet."
+)
+def test_create_with_too_many_devices(test_microvm_with_ssh, network_config):
+    """Test scenario: restore in previous versions with too many devices."""
+    test_microvm = test_microvm_with_ssh
+
+    # Create and start a microVM with `FC_V0_23_MAX_DEVICES_ATTACHED`
+    # network devices.
+    devices_no = FC_V0_23_MAX_DEVICES_ATTACHED
+    _create_and_start_microvm_with_net_devices(test_microvm,
+                                               network_config,
+                                               devices_no)
+
+    snapshot_builder = SnapshotBuilder(test_microvm)
+    # Create directory and files for saving snapshot state and memory.
+    _snapshot_dir = snapshot_builder.create_snapshot_dir()
+
+    # Pause microVM for snapshot.
+    response = test_microvm.vm.patch(state='Paused')
+    assert test_microvm.api_session.is_status_no_content(response.status_code)
+
+    # Attempt to create a full snapshot with version: `0.23.0`. Firecracker
```

Comment:
Please create a diff snapshot, it is faster :)

Reply:
Good idea. Done